### PR TITLE
Add System Event Models

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/12949

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    'WebSlotSwapWithPreviewCancelledEventData',
+    'WebSlotSwapWithPreviewStartedEventData',
+    'AppAction',
+    'AppServicePlanAction',
+    'AsyncStatus',
+    'MediaJobErrorCategory',
+    'MediaJobErrorCode',
+    'MediaJobRetry',
+    'MediaJobState',
+    'StampKind',
```
Comment (KieranBrantnerMagee) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/models/__init__.py:299]: Debating on whether we want this to be manual or not.  It seems like we could use some semblence of that importlib stanza I showed you?  Granted it seems... wrong to be doing automatic population of an import but I'm also not settled on the super-long-list of types.  I'd be curious to get @annatisch 's take on this as far as "python best practices" goes.

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py [hunk 2]
```diff
@@ -159,6 +159,17 @@ def model(self):
 
         :rtype: Union[CloudEvent, EventGridEvent]
         """
+        try:
+            event = self._event_dict['type'].rsplit('.')[-1]
```
Comment (KieranBrantnerMagee) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py:163]: is one of these for cloud_event vs egevent? let's leave helpful comments.

Comment (rakshith91) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py]: removed this - technically it's event_type for EGevent anf type for cloud event :(

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/models/__init__.py (hunk 1 above)
Comment (rakshith91) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/models/__init__.py:299]: you know my hatred towards long enumerations, but init makes exception IMO - would like to have more control over what we are exposing

Comment (KieranBrantnerMagee) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/models/__init__.py:299]: I'm fine to play this by ear.  My gut says in a few <months, years> as new services trickle in we'll end up wanting a less manual update process but that can very much be down the road.

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py [hunk 3]
```diff
@@ -176,14 +175,14 @@ def model(self):
...
-        Sets self._model.data to strongly typed event object if event type exists in _event_type_mappings.
+        Sets self._model.data to strongly typed event object if event type exists in _event_mappings.
         Otherwise, sets self._model.data to None.
 
         :param str event_type: The event_type of the EventGridEvent object or the type of the CloudEvent object.
         """
         # if system event type defined, set model.data to system event object
-        if event_type in DeserializedEvent._event_type_mappings:
-            self._model.data = (DeserializedEvent._event_type_mappings[event_type]).deserialize(self._model.data)
+        if event_type in _event_mappings:
```
Comment (KieranBrantnerMagee) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py:184]: God I feel VERY nitpicky saying this, but it's like the most "python idiom" thing ever; but try/except on keyError?  (One could feasibly make an argument the explicit if is more comprehensible but, ehh~.  I'm not gonna bikeshed on this more though so if you feel strongly etc.)

Comment (rakshith91) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py]: changed to try except - agree it's pythonic

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py [hunk 4]
```diff
@@ -176,15 +175,15 @@ def model(self):
...
 
         :param str event_type: The event_type of the EventGridEvent object or the type of the CloudEvent object.
         """
         # if system event type defined, set model.data to system event object
-        if event_type in DeserializedEvent._event_type_mappings:
-            self._model.data = (DeserializedEvent._event_type_mappings[event_type]).deserialize(self._model.data)
-        else:    # else, if custom event, then model.data is dict and should be set to None
+        try:
+            self._model.data = (_event_mappings[event_type]).deserialize(self._model.data)
+        except KeyError:
```
Comment (KieranBrantnerMagee) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py:186]:
```suggestion
        except KeyError: # else, if custom event, then model.data is dict and should be set to None
```

your comment was useful and I heartily support commenting :)

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py (hunk 2 above)
Comment (KieranBrantnerMagee) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py]: To be precise, was referring to the variable name (since it contains the type)

**Pull request merged**